echion/coremodule.cc

```diff
@@ -374,6 +375,81 @@ static PyObject* init_asyncio(PyObject* Py_UNUSED(m), PyObject* args)
     Py_RETURN_NONE;
 }
 
+// ----------------------------------------------------------------------------
+static PyObject* track_greenlet(PyObject* Py_UNUSED(m), PyObject* args)
+{
+    uintptr_t greenlet_id;  // map key
+    PyObject* name;
+    PyObject* frame_cell;
+
+    if (!PyArg_ParseTuple(args, "lOO", &greenlet_id, &name, &frame_cell))
+        return NULL;
+
+    StringTable::Key greenlet_name;
+
+    try
+    {
+        greenlet_name = string_table.key(name);
+    }
+    catch (StringTable::Error&)
+    {
+        // We failed to get this task but we keep going
+        PyErr_SetString(PyExc_RuntimeError, "Failed to get greenlet name from the string table");
+        return NULL;
+    }
+    {
+        const std::lock_guard<std::mutex> guard(greenlet_info_map_lock);
+
+        auto entry = greenlet_info_map.find(greenlet_id);
+        if (entry != greenlet_info_map.end())
+            // Greenlet is already tracked so we update its info
+            entry->second = std::make_unique<GreenletInfo>(greenlet_id, frame_cell, greenlet_name);
```

Comment:
Would it be possible to avoid creating a new `GreenletInfo` object by reusing it? How about having a `reset(greenlet_id, frame_cell, greenlet_name)`?

Reply:
🤔 In theory, we should never be hitting this code path as the only thing to update is the frame, and for that we use the indirection via a list object.